src/services/DiscordService.ts

```diff
@@ -0,0 +1,195 @@
+    }
+
+    client.login(TOKEN)
+  }
+
+  private static get channel() {
+    if (!CHANNEL_ID) {
+      throw new Error('Discord channel ID not set')
+    }
+
+    const channel = client.channels.cache.get(CHANNEL_ID)
+    if (channel?.type !== ChannelType.GuildText) {
+      throw new Error(`Discord channel type is not GuildText ${ChannelType.GuildText}`)
+    }
+
+    return channel
+  }
+
+  private static async formatMessage({
+    title,
+    proposalType,
+    description,
+    choices,
+    user,
+    action,
+    color,
+    url,
+  }: EmbedMessageProps) {
+    const fields: Field[] = []
+
+    if (!!proposalType && !!description) {
+      const embedDescription = !isGovernanceProcessProposal(proposalType)
+        ? description.split('\n')[0]
+        : shortText(description)
+
+      fields.push({
+        name: proposalType.toUpperCase().replaceAll('_', ' '),
+        value: embedDescription,
+      })
+    }
+
+    if (choices.length > 0) {
+      fields.push({ name: BLANK, value: BLANK }, ...choices)
+    }
+
+    const embed = new EmbedBuilder()
+      .setColor(color)
+      .setTitle(title)
+      .setURL(url)
+      .setDescription(action)
+      .setThumbnail('https://decentraland.org/images/decentraland.png')
+      .addFields(...fields)
+      .setTimestamp()
+      .setFooter({ text: 'Decentraland DAO', iconURL: 'https://decentraland.org/images/decentraland.png' })
+
+    if (user) {
+      const profile = await profiles.load(user)
+      const hasDclProfile = !!profile && !profile.isDefaultProfile
+      const profileHasName = hasDclProfile && !!profile.name && profile.name.length > 0
+      const displayableUser = profileHasName ? profile.name! : user
+
+      const hasAvatar = !!profile && !!profile.avatar
+
+      embed.setAuthor({
+        name: displayableUser,
+        iconURL: hasAvatar ? profile.avatar?.snapshots.face256 : DEFAULT_AVATAR,
+        url: profileUrl(user),
+      })
+    }
+
+    return embed
+  }
+
+  static async newProposal(
+    proposalId: string,
+    title: string,
+    type: ProposalType,
+    description: string,
+    choices: string[],
+    user: string
+  ) {
+    const action = 'A new proposal has been created'
+    const embedChoices = getChoices(choices)
+    const message = await this.formatMessage({
+      url: proposalUrl({ id: proposalId }),
+      title,
+      proposalType: type,
+      description,
+      choices: embedChoices,
+      user,
+      action,
+      color: MessageColors.NEW_PROPOSAL,
+    })
+    this.channel.send({ embeds: [message] })
+  }
+
+  static async newUpdate(proposalId: string, proposalTitle: string, updateId: string, user: string) {
+    const publicUpdates = getPublicUpdates(await UpdateModel.find<UpdateAttributes>({ proposal_id: proposalId }))
+    const updateIdx = Number(publicUpdates.findIndex((item) => item.id === updateId))
+    const updateNumber = publicUpdates.length > 0 && updateIdx >= 0 ? publicUpdates.length - updateIdx : NaN
```

Comment:
We have this logic somewhere else (maybe the proposal updates list and in the update detail view). Do you think we can reuse it somehow?